src/uwtools/tests/utils/test_file.py

```diff
@@ -24,6 +25,35 @@ def assets(tmp_path):
     return now, renamed, rundir
 
 
+def test_StdinProxy():
+    msg = "proxying stdin"
+    with patch.object(sys, "stdin", new=StringIO(msg)):
+        assert sys.stdin.read() == msg
+        # Reading from stdin a second time yields no input, as the stream has been exhausted:
+        assert sys.stdin.read() == ""
+    with patch.object(sys, "stdin", new=StringIO(msg)):
+        sp = file.StdinProxy()
+        assert sp.read() == msg
```

Comment:
Just want to confirm my understanding on these lines here. Using ```sys.stdin.read()``` causes the standard input to only be able to be read once, so when we read it again there is nothing to call here. Meaning it clears the cache after one use?

Once we use the ```StdinProxy()``` class this makes it so we can read the standard input multiple times hence why we have to clear the cache when we run each this each time i.e. in ```test_core.py```

Reply:
To be pedantic, using `sys.stdin.read()` doesn't _cause_ the standard input to only be able to be read once; the standard input is a stream which, be definition, can only be read once, regardless of what you do with it. You can think of it as a queue, with data being pushed on one end by some other process (e.g. by the `cat` command in something like `cat rocoto.xml | uw rocoto validate`, but it could also be a human typing on the keyboard), and then being popped off the other side (by `uw` reading it). But, yes, when we read it again, there's nothing there -- we've already read everything with the first `.read()`. No caching is involved in any of that.

The `StdinProxy` object reads (and exhausts) that stream one time, as usual, but stores it as a string in an instance variable. When the holder of a `StdinProxy` object calls `.read()` on it, `StdinProxy` streams that string to the caller in a way (mostly) indistinguishable from a real, live read from `stdin`. But users aren't expected to interact with `StdinProxy` objects directly; `readable()` handles that by calling the `_stdinproxy()` function, and it's this function that caches a single `StdinProxy` object, and its cache that needs to be reset so that a new `StdinProxy` object, which reads a new (mocked) `stdin` stream created by a unit test, can be created. Clearing the cache shouldn't be necessary during normal `uwtools` use, as we wouldn't normally read `stdin` more than once. But the unit tests need to.